aten/src/ATen/native/nested/NestedTensorMath.cpp

```diff
@@ -869,14 +869,16 @@ _NestedTensor_GeneralizedBMM_BatchSizes_OutputMemory(
 }
 }
 
-// This is a generalized batched matmul dedicated to nested tensors,
+// Note [nested tensor matmul] TODO
+// This is really a generalized batched matmul dedicated to nested tensors,
 // where `self` and `mat2` have same number (>= 3) of dimensions.
 // The last 2 dimensions will be considered as matrix dimensions,
 // so they should be matrix-multiplicable.
 // The leading dimensions are considered as batch dimensions,
 // and since nested tensor does not support broadcasting for now,
 // for each batch dimension `self` and `mat2` must have same size.
-Tensor _NestedTensor_GeneralizedBMM(const Tensor& self, const Tensor& mat2) {
+// Should make full matmul semantics support some day
```

Comment:
What does that mean? There are cases where we have errors and not others?

Reply:
Yes. For now only the "generalized bmm" case is supported. Other cases (dot, mv, mm) are errored out.

There was a discussion among me and @jbschlosser and @cpuhrsch on whether to support this partially functional nested matmul. For transformer purpose, we could either support reshaping nested dimension 0 to use bmm, or make reshape simple but add this matmul.

Reply:
I think it is fine as long as you get a very clear error for the things not supported.